assets/js/Steps/CheckInItems.js

```diff
@@ -31,74 +33,75 @@ import Input from './components/input';
 function CheckInItems({ actionHandler }) {
     const context = useContext(MachineStateContext);
     const [scannedBarcode, setScannedBarcode] = useState('');
-    const [infoString, setInfoString] = useState('');
 
     /**
      * Set up barcode scanner listener.
      */
     useEffect(() => {
-        // @TODO: Why this call?
-        setInfoString(
-            scannedBarcode ? 'Bogen blev registreret. Klar til næste' : ''
-        );
-
         const barcodeScanner = new BarcodeScanner(BARCODE_SCANNING_TIMEOUT);
 
         const barcodeCallback = (code) => {
+            const whichFlow = context.machineState.get.user
+                ? 'changeFlow'
+                : 'enterFlow';
             if (code.length === BARCODE_COMMAND_LENGTH) {
-                if (code === BARCODE_COMMAND_FINISH) {
-                    actionHandler('changeFlow', { flow: 'reset' });
-                }
-                const whichFlow = context.machineState.get.user
-                    ? 'changeFlow'
-                    : 'enterFlow';
-                if (code === BARCODE_COMMAND_STATUS) {
-                    actionHandler(whichFlow, {
-                        flow: 'status'
-                    });
-                }
-
-                if (code === BARCODE_COMMAND_CHECKOUT) {
-                    actionHandler(whichFlow, {
-                        flow: 'checkOutItems'
-                    });
+                switch (code) {
+                    case BARCODE_COMMAND_FINISH:
+                        actionHandler('reset');
+                        break;
+                    case BARCODE_COMMAND_STATUS:
+                        actionHandler(whichFlow, {
+                            flow: 'status',
+                        });
+                        break;
+                    case BARCODE_COMMAND_CHECKIN:
+                        actionHandler(whichFlow, {
+                            flow: 'checkOutItem',
+                        });
+                        break;
+                    default:
+                        actionHandler('checkInItem', {
+                            itemIdentifier: code,
+                        });
                 }
-                return;
+                setScannedBarcode(code);
             }
-            setScannedBarcode(code);
-            actionHandler('checkInItem', {
-                itemIdentifier: code
-            });
         };
 
         barcodeScanner.start(barcodeCallback);
         return () => {
             barcodeScanner.stop();
         };
     }, [actionHandler]);
```

Comment:
It now checks for the items (at some point the state machine send [] and not nothing, I think?)